lib/new_relic/tracer.ex

```diff
@@ -58,6 +58,39 @@ defmodule NewRelic.Tracer do
     send_resp(conn, 200, "ok")
   end
   ```
+
+  ###### Customize external traces
+
+  Its possible to report external trace with custom names.
+
+  To change External Service calls names, use this syntax:
+
+  ```elixir
+  defmodule MyExternalService do
+    use NewRelic.Tracer
+
+    @trace {:query, category: :external, reported_name: :external_host}
+    def query(args) do
+      # Make the call
+    end
+
+    def external_host(_args), do: "external-domain.net"
+  end
+  ```
+
+  Alternatively it also can be set with a short and a long name:
+
+  ```elixir
+  defmodule MyExternalService do
+    use NewRelic.Tracer
+
+    @trace {:query, category: :external, reported_name_tuple: {"/users", "some-domain.net/users"}}
```

Comment:
Regardless of category, it looks like NR will always track the original function name, then the given `name` (first element of this `@trace` tuple) will work as a static override for displaying purposes. With that in mind, I think the most flexible way to approach this is by allowing `name` to be an MF too, making it dynamic, like:

```elixir
defmodule MyApp.MyMod do
  @trace {:custom_name, category: :external}
  def query(arg) do
    # Reports as MyApp.MyMod.query:custom_name/1
  end

  @trace {{MyHttpLibAdapter, :name}, category: :external}
  def query2(arg) do
    # Reports as MyApp.MyMod.query2:[METHOD]the-host.tld/path/1
  end
end

defmodule MyHttpLibAdapter do
  def name(args) do
    # Generate string to be reported based on same args as the function being monitored.
    "[METHOD]the-host.tld/path"
  end
end
```

That way we still use `name` for its original purpose and don't need to create another option for the `@trace` tag. One caveat though: we might need to make some atom2string changes in the tracking code to avoid memory leaks, now that names can't be determined at compile time.